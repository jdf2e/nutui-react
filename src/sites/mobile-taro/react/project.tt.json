{
  "miniprogramRoot": "./",
  "projectname": "first",
  "appid": "testAppId",
  "setting": {
    "es6": false,
    "minified": false
  }
}
