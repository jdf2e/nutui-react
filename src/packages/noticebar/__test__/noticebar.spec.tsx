import * as React from 'react'
import { render, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom'
import NoticeBar from '@/packages/noticebar'

test('noticebar base test', () => {
  const text =
    'NutUI 是京东风格的移动端组件库，使用 Vue 语言来编写可以在 H5，小程序平台上的应用，帮助研发人员提升开发效率，改善开发体验。'

  const { container } = render(<NoticeBar content={text} />)
  expect(
    container.querySelector('.nut-noticebar-box-wrap-content')?.innerHTML
  ).toBe(
    'NutUI 是京东风格的移动端组件库，使用 Vue 语言来编写可以在 H5，小程序平台上的应用，帮助研发人员提升开发效率，改善开发体验。'
  )

  expect(container).toMatchSnapshot()
})

test('scrollable test', () => {
  const text =
    'NutUI 是京东风格的移动端组件库，使用 Vue 语言来编写可以在 H5，小程序平台上的应用，帮助研发人员提升开发效率，改善开发体验。'

  const { container } = render(<NoticeBar content={text} scrollable={false} />)
  setTimeout(() => {
    expect(
      container.querySelector('.nut-noticebar-box-wrap-content')
    ).toHaveClass('nut-ellipsis')
    expect(
      container.querySelector('.nut-noticebar-box-wrap-content')
    ).toHaveAttribute(
      'style',
      'animation-delay: 1s; animation-duration: 0s; transform: translateX(0);'
    )
  }, 300)

  expect(container).toMatchSnapshot()
})

test('closeable & rightIcon test', () => {
  const text =
    'NutUI 是京东风格的移动端组件库，使用 Vue 语言来编写可以在 H5，小程序平台上的应用，帮助研发人员提升开发效率，改善开发体验。'
  const handleClick = jest.fn()
  const { container } = render(
    <NoticeBar
      content={text}
      closeable
      onClick={handleClick}
      rightIcon="circle-close"
    />
  )
  setTimeout(() => {
    expect(container.querySelector('.nut-noticebar-box-right-icon')).toBeTruthy
    expect(container.querySelector('.nutui-iconfont')).toHaveClass(
      'nut-icon-circle-close'
    )
    fireEvent.click(container)
    expect(handleClick).toBeCalled()
  }, 300)
})

test('customer leftIcon test', () => {
  const { container } = render(
    <NoticeBar leftIcon="https://img13.360buyimg.com/imagetools/jfs/t1/72082/2/3006/1197/5d130c8dE1c71bcd6/e48a3b60804c9775.png">
      <a href="https://www.jd.com">京东商城</a>
    </NoticeBar>
  )
  setTimeout(() => {
    expect(container.querySelector('.nut-noticebar-box-left-icon')).toBeTruthy
    expect(
      container.querySelector('.nut-noticebar-box-left-icon')
    ).toHaveAttribute(
      'style',
      'background-image: url("https://img13.360buyimg.com/imagetools/jfs/t1/72082/2/3006/1197/5d130c8dE1c71bcd6/e48a3b60804c9775.png");'
    )
    expect(
      container.querySelector('.nut-noticebar-box-wrap-content')?.innerHTML
    ).toBe('<a href="https://www.jd.com">京东商城</a>')
  }, 300)
})

test('wrap test', () => {
  const text =
    'NutUI 是京东风格的移动端组件库，使用 Vue 语言来编写可以在 H5，小程序平台上的应用，帮助研发人员提升开发效率，改善开发体验。'
  const { container } = render(<NoticeBar content={text} wrap />)
  expect(container.querySelector('.nut-noticebar-box')).toHaveClass(
    'nut-noticebar-box-wrapable'
  )
})

test('vertical test', () => {
  const horseLamp1 = [
    'NoticeBar 公告栏',
    'Cascader 级联选择',
    'DatePicker 日期选择器',
    'CheckBox 复选按钮',
  ]
  const { container } = render(
    <NoticeBar
      direction="vertical"
      list={horseLamp1}
      speed={10}
      duration={1000}
      closeable
    />
  )
  expect(container.querySelector('.nut-noticebar-vertical')).toBeTruthy
  expect(container.querySelector('.nut-noticebar-box-horseLamp_list'))
    .toBeTruthy
})

test('vertical test', () => {
  const horseLamp1 = [
    'NoticeBar 公告栏',
    'Cascader 级联选择',
    'DatePicker 日期选择器',
    'CheckBox 复选按钮',
  ]
  const { container } = render(
    <NoticeBar direction="vertical" height={50} speed={10} duration={1000}>
      {horseLamp1.map((item, index) => {
        return (
          <div
            className="custom-item"
            style={{ height: '50px', lineHeight: '50px' }}
            key={index}
          >
            {item}
          </div>
        )
      })}
    </NoticeBar>
  )
  expect(container.querySelector('.custom-item')).toBeTruthy
})
