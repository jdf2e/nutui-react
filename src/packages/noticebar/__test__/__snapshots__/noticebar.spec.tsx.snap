// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`noticebar base test 1`] = `
<div>
  <div
    class="nut-noticebar "
  >
    <div
      class="nut-noticebar-box"
    >
      <div
        class="left-icon"
      >
        <svg
          aria-labelledby="Notice"
          class="nut-icon nut-icon-Notice "
          role="presentation"
          style="width: 16px; height: 16px;"
          viewBox="0 0 1027 1024"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M940.873 819.2h-85.334V430.933c0-192-153.6-345.6-341.333-345.6s-341.333 153.6-341.333 345.6V819.2H87.539V430.933C87.54 192 279.54 0 514.206 0s426.667 192 426.667 430.933V819.2zM514.633 1024c-68.267 0-132.267-29.867-174.934-85.333-17.066-17.067-12.8-42.667 4.267-59.734 17.067-12.8 46.933-12.8 59.733 8.534 29.867 34.133 68.267 51.2 110.934 51.2s81.066-17.067 106.666-51.2c12.8-17.067 42.667-21.334 59.734-8.534 17.066 12.8 21.333 42.667 8.533 59.734C646.899 994.133 582.899 1024 514.633 1024zm469.333-162.133h-12.8c-294.4-89.6-618.667-89.6-913.067 0-21.333 4.266-46.933-8.534-55.466-29.867-8.534-21.333 4.266-46.933 29.866-51.2 311.467-93.867 652.8-89.6 964.267 0 21.333 8.533 34.133 29.867 29.867 51.2-8.534 21.333-25.6 29.867-42.667 29.867z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </div>
      <div
        class="wrap"
      >
        <div
          class="content  "
          style="animation-delay: 1s; animation-duration: 0s; transform: translateX(0);"
        >
          NutUI 是京东风格的移动端组件库，使用 Vue 语言来编写可以在 H5，小程序平台上的应用，帮助研发人员提升开发效率，改善开发体验。
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`scrollable test 1`] = `
<div>
  <div
    class="nut-noticebar "
  >
    <div
      class="nut-noticebar-box"
    >
      <div
        class="left-icon"
      >
        <svg
          aria-labelledby="Notice"
          class="nut-icon nut-icon-Notice "
          role="presentation"
          style="width: 16px; height: 16px;"
          viewBox="0 0 1027 1024"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M940.873 819.2h-85.334V430.933c0-192-153.6-345.6-341.333-345.6s-341.333 153.6-341.333 345.6V819.2H87.539V430.933C87.54 192 279.54 0 514.206 0s426.667 192 426.667 430.933V819.2zM514.633 1024c-68.267 0-132.267-29.867-174.934-85.333-17.066-17.067-12.8-42.667 4.267-59.734 17.067-12.8 46.933-12.8 59.733 8.534 29.867 34.133 68.267 51.2 110.934 51.2s81.066-17.067 106.666-51.2c12.8-17.067 42.667-21.334 59.734-8.534 17.066 12.8 21.333 42.667 8.533 59.734C646.899 994.133 582.899 1024 514.633 1024zm469.333-162.133h-12.8c-294.4-89.6-618.667-89.6-913.067 0-21.333 4.266-46.933-8.534-55.466-29.867-8.534-21.333 4.266-46.933 29.866-51.2 311.467-93.867 652.8-89.6 964.267 0 21.333 8.533 34.133 29.867 29.867 51.2-8.534 21.333-25.6 29.867-42.667 29.867z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
      </div>
      <div
        class="wrap"
      >
        <div
          class="content  "
          style="animation-delay: 1s; animation-duration: 0s; transform: translateX(0);"
        >
          NutUI 是京东风格的移动端组件库，使用 Vue 语言来编写可以在 H5，小程序平台上的应用，帮助研发人员提升开发效率，改善开发体验。
        </div>
      </div>
    </div>
  </div>
</div>
`;
